tests/test_models/deterministic.py

```diff
@@ -0,0 +1,171 @@
+"""Example specifications of a deterministic consumption-saving model.
+
+This specification builds on the example model presented in the paper: "The endogenous
+grid method for discrete-continuous dynamic choice models with (or without) taste
+shocks" by Fedor Iskhakov, Thomas H. Jørgensen, John Rust and Bertel Schjerning (2017,
+https://doi.org/10.3982/QE643).
+
+"""
+
+import jax.numpy as jnp
+
+# ======================================================================================
+# Numerical parameters and constants
+# ======================================================================================
+
+N_GRID_POINTS = {
+    "wealth": 100,
+    "consumption": 500,
+}
+
+RETIREMENT_AGE = 65
+
+# ======================================================================================
+# Model functions
+# ======================================================================================
+
+
+# --------------------------------------------------------------------------------------
+# Utility functions
+# --------------------------------------------------------------------------------------
+def utility(consumption, working, disutility_of_work):
+    return jnp.log(consumption) - disutility_of_work * working
+
+
+def utility_with_filter(
+    consumption,
+    working,
+    disutility_of_work,
+    # Temporary workaround for bug described in issue #30, which requires us to pass
```

Comment:
We should enable the todo stuff from ruff, which enforces a format like:

    # TODO(@hmgaudecker): Remove explicit cast when vectorisation is enabled.
    # https://github.com/iza-institute-of-labor-economics/gettsim/issues/515

I find that quite useful, at least.